File: dcp-solutions/2025-01-31/solution.js
```javascript
function partition(lst, x) {
    let less = [];
    let equal = [];
    let greater = [];

    lst.forEach(num => {
        if (num < x) {
            less.push(num);
        } else if (num === x) {
            equal.push(num);
        } else {
            greater.push(num);
        }
    });

    return [less, equal, greater];
}
```

File: dcp-solutions/2025-01-31/solution.go
```go
func partition(lst []int, x int) [][]int {
    less := make([]int, 0)
    equal := make([]int, 0)
    greater := make([]int, 0)

    for _, num := range lst {
        if num < x {
            less = append(less, num)
        } else if num == x {
            equal = append(equal, num)
        } else {
            greater = append(greater, num)
        }
    }

    return [][][]int{{less, equal, greater}}
}
```